recipes/cubicinterpolation/all/conanfile.py

```diff
@@ -0,0 +1,98 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+
+class CubicInterpolationConan(ConanFile):
+    name = "cubicinterpolation"
+    homepage = "https://github.com/MaxSac/cubic_interpolation"
+    license = "MIT"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "Leightweight interpolation library based on boost and eigen."
+    topics = ("interpolation", "splines", "cubic", "bicubic", "boost", "eigen3")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {"shared": False, "fPIC": True}
+    generators = "cmake"
+    _source_subfolder = "source_subfolder"
+    exports_sources = ["CMakeLists.txt"]
+    _cmake = None
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    @property
+    def _minimum_compilers_version(self):
+        return {
+            "Visual Studio": "15",
+            "gcc": "5",
+            "clang": "5",
+            "apple-clang": "5.1",
+        }
+
+    def validate(self):
+        if self.settings.compiler.cppstd:
+            check_min_cppstd(self, "14")
+
+        minimum_version = self._minimum_compilers_version.get(
+            str(self.settings.compiler), False
+        )
+        if not minimum_version:
+            self.output.warn(
+                "CubicInterpolation requires C++14. Your compiler is unknown. Assuming it supports C++14."
+            )
+        elif tools.Version(self.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration(
+                "CubicInterpolation requires C++14, which your compiler does not support."
+            )
+
+        if self.settings.os == "Windows":
+            raise ConanInvalidConfiguration("This library is not compatible with Windows")
+
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = "cubic_interpolation-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def requirements(self):
+        self.requires("boost/1.75.0")
+        self.requires("eigen/3.3.9")
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
```

Comment:
```suggestion
        self._cmake = CMake(self)
        self._cmake["BUILD_EXAMPLE"] = False
        self._cmake["BUILD_DOCUMENTATION"] = False
```

Already the default values, but it's better to be explicit.